### ISSUE TITLE
dsymutil warning

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#15594]
updating command line tools gains us a new build warning and probably failure mode:
```
~/julia/build-llvm37$ make -C deps
dsymutil: Not enough positional command line arguments specified!
Must specify at least 1 positional arguments: See: /Library/Developer/CommandLineTools/usr/bin/dsymutil -help
/bin/sh: line 0: test: -gt: unary operator expected
make: Nothing to be done for `default'.
```

indeed, it looks like apple is now bundling llvm's dsymutil (i think @keno will like that), which wants a different invocation command:
```
~/julia$ dsymutil -v -
LLVM (http://llvm.org/):
  LLVM version 7.3.0
  Optimized build.
  Built Mar  8 2016 (10:26:42).
  Default target: x86_64-apple-darwin15.3.0
  Host CPU: broadwell
~/julia$ dsymutil --version
LLVM (http://llvm.org/):
  LLVM version 7.3.0
  Optimized build.
  Built Mar  8 2016 (10:26:42).
  Default target: x86_64-apple-darwin15.3.0
  Host CPU: broadwell
```

**Comment by Keno** [comment]
Great, that means we can also switch over the DWARF version to 4 by default on all platforms.

**Comment by tkelman** [comment]
Do they backport new command line tools releases to all old os x versions?

**Comment by slarew** [comment]
@tkelman Looking at the Apple Developer [downloads page](https://developer.apple.com/downloads/), the trend appears to be to supply the most recent command line tools for the current and (sometimes) previous OS X releases.  If the previous OS is supported, install packages for both OS releases are released nearly simultaneously.  By late Spring, the previous OS is no longer supported.

That said, the newest command line tools (for Xcode 7.3) has only been released for 10.11.  Based on the observed trend, I'm skeptical that a 10.10 backport will be released.
